plutus-ledger/src/Ledger/Index/Internal.hs

```diff
@@ -31,43 +36,45 @@ newtype UtxoIndex = UtxoIndex { getIndex :: Map.Map TxOutRef TxOut }
     deriving newtype (Eq, Semigroup, OpenApi.ToSchema, Monoid, Serialise)
     deriving anyclass (FromJSON, ToJSON, NFData)
 
+-- | Create an index of all UTxOs on the chain.
+initialise :: Blockchain -> UtxoIndex
+initialise = UtxoIndex . unspentOutputs
+
+instance Eq (C.Ledger.BasicFailure c) where
+    _ == _ = True
+instance Eq (C.Ledger.ScriptFailure c) where
+    _ == _ = True
+
+instance ToJSON (C.Ledger.BasicFailure c) where
+    toJSON _ = toJSON "BasicFailure"
+instance FromJSON (C.Ledger.BasicFailure c) where
+    parseJSON = error "not implemented"
+instance ToJSON (C.Ledger.ScriptFailure c) where
+    toJSON _ = toJSON "ScriptFailure"
+instance FromJSON (C.Ledger.ScriptFailure c) where
+    parseJSON = error "not implemented"
+
+instance ToJSON (C.Ledger.ApplyTxError EmulatorEra) where
+    toJSON _ = toJSON "ApplyTxError"
+instance FromJSON (C.Ledger.ApplyTxError EmulatorEra) where
+    parseJSON = error "not implemented"
+
+instance ToJSON (C.Ledger.UtxosPredicateFailure EmulatorEra) where
+    toJSON _ = toJSON "UtxosPredicateFailure"
+instance FromJSON (C.Ledger.UtxosPredicateFailure EmulatorEra) where
+    parseJSON = error "not implemented"
```

Comment:
Actually, we can save a lot of code by getting back `CardanoLedgerValidationError String`. We won't need to write the instances and we still use `isInfixOf` to check the contents of error message.

I just tried to get rid of `CardanoLedgerValidationError` because we now can use cardano-ledger properly with their tests but it's annoying.

So please share you opinion.

Reply:
I just tried to deal with aeson instances and got annoyed a lot by lots of missing instances for cardano types. So I'm getting `CardanoLedgerValidationError String` back for now.